doc/tutorials/tutorial_understanding_building_benchmark.rst

```diff
@@ -0,0 +1,370 @@
+
+.. prompt:: bash $
+
+    cp python-gd.py python-gd-ridge.py
+
+You should now have a second file in the ``solvers/`` directory.
+Since Benchopt easily runs several algorithms and compares them natively, we will be able to see if adapting the GD algorithm to the ridge problem is actually useful or not!
+
+Modifying the solver means updating the ``run`` method, more specifically the gradient formula.
+Inside the ``python-gd-ridge.py`` file, the new ``run`` method looks like this
+
+.. code-block:: python
+
+    def run(self, n_iter):
+        L = np.linalg.norm(self.X, ord=2) ** 2
+        alpha = self.scale_step / L
+        beta = np.zeros(self.X.shape[1])
+        for _ in range(n_iter):
+            beta -= alpha * (gradient_ols(self.X, self.y, beta) + 2*self.reg*beta)
+
+        self.beta = beta
+
+Note that we are using ``self.reg`` as the value of \\( \\lambda \\).
+To get this value from the ``objective.py`` file, we need to update the ``set_objective`` method, which is the counterpart of ``get_objective`` we just updated in ``objective.py``.
+
+.. code-block:: python
+
+    def set_objective(self, X, y, reg):
+        self.X, self.y, self.reg = X, y, reg
+
+As a final step, we can change the name of our new algorithm to ``"GD-ridge"`` in the class attribute ``name``, to differentiate it from ``"GD"`` in the visualization of the results.
+
+
+.. code-block:: python
+
+    class Solver(BaseSolver):
+        name = 'GD_ridge'
+
+
+And that's it, you now have your first benchmark setup! Congratulations :)
+
+All that's left is to run the benchmark and look at the results.
+We run the benchopt with the same command as earlier, in the parent directory of the template benchmark:
+
+.. prompt:: bash $
+
+    benchopt run my_benchmark
+
+Notice how the prompt in the terminal now contains logging for the two algorithms ``GD`` and ``GD-ridge``.
+
+.. image:: console_ridge.png
+    :width: 600
+
+Upon completion of the run, you should again have all the results collected in a new interactive window in your default web navigator.
+There is a lot of interesting information contained in these results.
+For instance, select the following plotting options:
+
+    - Simulated[n_features=500, n_samples=1000]
+    - Ordinary Least Squares[reg=100.0, whiten_y=False]
+    - objective_value
+    - suboptimality_curve
+    - loglog
+    - iteration
+
+You should see the following plot 
+
+.. image:: screenshot_mytemplate_ridgeloss.png
+    :width: 800
+
+We may observe that the GD-ridge with ``stepsize=1`` reaches a much lower cost value that other methods, in particular GD without the ridge penalisation term.
+This is expected since GD does not minimizes \\(f(\\beta) \\) but rather  \\( g(\\beta) \\).
+Interestingly, using ``stepsize=1.99`` leads to large errors.
+This is caused by the scaling of the stepsize (using the Lipschitz constant) not being updated in our GD-ridge algorithm, and the stepsize is therefore not safe.
```

Comment:
Too complex and it gives me (my 2 cents) the impression that benchopt is buggy.

I strongly prefer a selling picture where everything is correct and works fine

Reply:
I will improve on that, stick to ridge only as suggested.